tests/queries/0_stateless/01744_alter_replace_partition_update.sql

```diff
@@ -0,0 +1,66 @@
+
+-- replace from existing partition to non-existing partition
+alter table x replace partition ('daily', '2021-03-29 00:00:00') update type = 'daily' in partition ('hourly', '2021-03-29 06:00:00');
+
+-- replace from non-existing partition to non-existing partition
+alter table x replace partition ('daily', '2021-03-29 00:00:00') update type = 'daily' in partition ('hourly', '2021-03-09 06:00:00');
+
+-- replace from non-existing partition to existing partition
+-- replace will do nothing in this case
+alter table x replace partition ('daily', '2021-03-28 00:00:00') update type = 'daily' in partition ('hourly', '2021-03-09 06:00:00');
+
+-- attach partition
+alter table x attach partition ('daily', '2021-03-28 00:00:00') update type = 'daily' in partition ('hourly', '2021-03-28 08:00:00');
+
+select * from x order by type, dt, i;
+
+create table x1 (type Enum8('hourly' = 0, 'hourly_staging' = 1, 'daily' = 2, 'daily_staging' = 3), dt DateTime, i int) engine ReplicatedMergeTree('/clickhouse/tables/test_01744/01/x', '1') partition by (type, if(toUInt8(type) < 2, dt, toStartOfDay(dt))) order by i;
+
+create table x2 (type Enum8('hourly' = 0, 'hourly_staging' = 1, 'daily' = 2, 'daily_staging' = 3), dt DateTime, i int) engine ReplicatedMergeTree('/clickhouse/tables/test_01744/01/x', '2') partition by (type, if(toUInt8(type) < 2, dt, toStartOfDay(dt))) order by i;
+
+insert into x1 values ('daily', '2021-03-28 01:00:00', 10), ('daily', '2021-03-28 02:00:00', 20);
+insert into x1 values ('hourly', '2021-03-28 05:00:00', 60), ('hourly', '2021-03-28 06:00:00', 70);
+insert into x1 values ('hourly', '2021-03-28 07:00:00', 80), ('hourly', '2021-03-28 08:00:00', 90);
+insert into x1 values ('hourly', '2021-03-29 05:00:00', 60), ('hourly', '2021-03-29 06:00:00', 70);
+
+select * from x1 order by type, dt, i;
+
+-- replace from existing partition to existing partition
+alter table x1 replace partition ('daily', '2021-03-28 00:00:00') update type = 'daily' in partition ('hourly', '2021-03-28 07:00:00');
+
+-- replace from existing partition to non-existing partition
+alter table x1 replace partition ('daily', '2021-03-29 00:00:00') update type = 'daily' in partition ('hourly', '2021-03-29 06:00:00');
+
+-- replace from non-existing partition to non-existing partition
+alter table x1 replace partition ('daily', '2021-03-29 00:00:00') update type = 'daily' in partition ('hourly', '2021-03-09 06:00:00');
+
+-- replace from non-existing partition to existing partition
+-- replace will do nothing in this case
+alter table x1 replace partition ('daily', '2021-03-28 00:00:00') update type = 'daily' in partition ('hourly', '2021-03-09 06:00:00');
+
+-- attach partition
+alter table x1 attach partition tuple('daily', '2021-03-28 00:00:00') update type = 'daily' in partition ('hourly', '2021-03-28 08:00:00');
+
+select * from x1 order by type, dt, i;
+
+system sync replica x2;
+
+select * from x2 order by type, dt, i;
+
```

Comment:
Please, add the tests for all possible failure situations, like mismatched partition, or attempt to replace multiple partitions.

Reply:
Done.